src/engine/surface_collision.c

```diff
@@ -299,64 +338,67 @@ static s32 check_within_ceil_triangle_bounds(s32 x, s32 z, struct Surface *surf,
  * Iterate through the list of ceilings and find the first ceiling over a given point.
  */
 static struct Surface *find_ceil_from_list(struct SurfaceNode *surfaceNode, s32 x, s32 y, s32 z, f32 *pheight) {
-    register struct Surface *surf, *ceil = NULL;
-    register f32 height;
-    SurfaceType type = SURFACE_DEFAULT;
-    *pheight = CELL_HEIGHT_LIMIT;
-    // Stay in this loop until out of ceilings.
+    struct Surface *ceil = NULL;
+    struct Surface *lowestCeil = NULL;
+    f32 lowestCeilHeight = *pheight;
+
+    _Bool checkingForCamera = (gCollisionFlags & COLLISION_FLAG_CAMERA);
```

Comment:
I changed them to s32. Later we should start using `_Bool` though, It's a u8 that enforces being either 0 or 1 with an automatic `x != 0` so it's useful especially when checking bitfields. Another example in the repo where that matters is in `geo_process_master_list_sub` where `enableZBuffer` is used as an index in the render mode cycle arrays and would break if it didn't have the `!= 0`.